src/web/routes/application/common/state-machine/state-machine.js

```diff
@@ -17,9 +17,33 @@ const actions = {
 
 const getStepForPath = (path, steps) => steps.find(step => step.path === path)
 
+/**
+ * Next path is navigable if the next step does not exist, if it doesn't define an isNavigable function, or that function returns true.
+ * E.g. if the current step is the end of the 'in-progress' journey, the next path will be /check. There is no step matching /check, so we assume it is navigable.
+ */
+function isNextPathNavigable (nextStep, req) {
+  return !nextStep || !nextStep.isNavigable || nextStep.isNavigable(req.session)
```

Comment:
Maybe unnecessary but `isNil(nextStep) || isNil(nextStep.isNavigable)` is more explicit than `!nextStep ...`.

`nextStep.isNavigable(req.session)` is the only check where we are expecting a boolean.